managed/CounterStrikeSharp.API/Modules/Admin/AdminManager.cs

```diff
@@ -0,0 +1,173 @@
+﻿using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Text.Json.Serialization;
+using System.Text.Json;
+using CounterStrikeSharp.API.Core;
+using CounterStrikeSharp.API.Modules.Utils;
+using CounterStrikeSharp.API.Modules.Entities;
+using CounterStrikeSharp.API.Modules.Commands;
+using System.Linq;
+using System.Reflection;
+
+namespace CounterStrikeSharp.API.Modules.Admin
+{
+    public partial class AdminData
+    {
+        [JsonPropertyName("auth_type")] public string AuthType { get; set; }
+        [JsonPropertyName("identity")] public string Identity { get; set; }
+        [JsonPropertyName("flags")] public List<string> Flags { get; set; }
```

Comment:
Does System.Text.Json handle deserialisation to a `HashSet`, if it does this would be a better data structure for this for quick flag lookup.

Reply:
Didn't work from my testing unfortunately. Might be something we come back to in the future.